lib/Backend/LowerMDSharedSimd128.cpp

```diff
@@ -771,8 +771,8 @@ IR::Instr* LowererMD::Simd128LowerLdLane(IR::Instr *instr)
 
         // cmp      dst, -1
         pInstr = IR::Instr::New(Js::OpCode::CMP, m_func);
-        pInstr->SetSrc1(dst);
-        pInstr->SetSrc2(IR::IntConstOpnd::New(-1, TyInt32, m_func, true));
+        pInstr->SetSrc1(dst->UseWithNewType(laneType, m_func));
```

Comment:
Better to avoid UseWithNewType. In this case, Legalize(pInstr) will use the smallest IRType in the MOV instruction, which is laneType. So we can remove UseWithNewType here.